edx_exams/apps/router/middleware.py

```diff
@@ -0,0 +1,32 @@
+"""
+Middleware that checks if in incoming request should be
+routed to the legacy proctoring service (edx-proctoring)
+"""
+from django.utils.deprecation import MiddlewareMixin
+
+from edx_exams.apps.api.v1.views import CourseExamsView
+from edx_exams.apps.core.models import CourseExamConfiguration
+from edx_exams.apps.router.views import CourseExamsOverrideView
+
+OVERRIDE_VIEW_MAP = {
+    CourseExamsView: CourseExamsOverrideView
+}
+
+class ExamRequestMiddleware(MiddlewareMixin):
+    """
+    Intercept requests and determine if exams for this course
+    should be managed by legacy edx-proctoring system. If so, requests
+    are intercepted and forwarded to that service
+    """
+    def process_view(self, request, view_func, view_args, view_kwargs):
+        try:
+            override_view = OVERRIDE_VIEW_MAP.get(view_func.view_class)
+        except AttributeError:
+            override_view = None
+
+        # call into override if a function is defined for this request method
+        if override_view and getattr(override_view, request.method.lower()):
```

Comment:
the getattr is always a bad sign as is the fact that you could invisibly have a POST override but no PATCH override

Would it make sense to require that the override views cover the same methods as the originals? And just let things fail messily if they don't.

Or make the map be method specific?
(CourseExamsView, PATCH): CourseExamsOverrideView
after writing that out I really don't like it

Reply:
yeah I was trying to have this fall through seamlessly but now that you point it out that's pretty undesirable. I can see no reason we would ever want GET to go one place and PATCH to go another 😬